src/mono/mono/utils/mono-threads-wasm.c

```diff
@@ -411,6 +411,27 @@ mono_threads_platform_stw_defer_initial_suspend (MonoThreadInfo *info)
 	return mono_native_thread_id_equals (mono_thread_info_get_tid (info), mono_threads_wasm_browser_thread_tid ());
 }
 
+#ifndef DISABLE_THREADS
+extern void
+mono_wasm_pthread_on_pthread_created (gpointer pthread_id);
+#endif
+
+void
+mono_threads_wasm_on_thread_attached (void)
+{
+#ifdef DISABLE_THREADS
+	return;
+#else
+	if (mono_threads_wasm_is_browser_thread ()) {
+		return;
+	}
+	// Set up a MessageChannel between the new thread (which might be on a pooled reused WebWorker) and the main thread.
+	pthread_t id = pthread_self ();
+	mono_wasm_pthread_on_pthread_created (id);
```

Comment:
```suggestion
	MONO_ENTER_GC_SAFE;
	mono_wasm_pthread_on_pthread_created (id);
	MONO_EXIT_GC_SAFE;
```

Hmm. I probably should go back through my earlier PRs and make sure I'm doing a GC Safe transition on exits from C to JS...